File: SIH/cyber-risk-platform/backend/app/schemas/recommendation.py
import uuid
from pydantic import BaseModel, ConfigDict
from datetime import datetime
from typing import Any

class RecommendationRead(BaseModel):
    id: uuid.UUID
    organization_id: uuid.UUID
    asset_id: uuid.UUID | None = None
    title: str
    description: str | None = None
    priority: str
    estimated_cost: float | None = None
    expected_risk_reduction: float | None = None
    status: str
    generated_at: datetime
    accepted_at: datetime | None = None
    completed_at: datetime | None = None
    metadata_: dict[str, Any] | None = None
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True, populate_by_name=True)

File: SIH/cyber-risk-platform/frontend/src/types/recommendation.ts
export interface RecommendationEvidence {
  source: string;
  detail: string;
  severity: string;
}

export interface RecommendationMetadata {
  rationale: string;
  risk_driver: string;
  urgency: string;
  expected_financial_benefit?: number;
  implementation_effort: string;
  confidence: number;
  evidence: RecommendationEvidence[];
}

export interface Recommendation {
  id: string;
  organization_id: string;
  asset_id?: string;
  title: string;
  description?: string;
  priority: string;
  estimated_cost?: number;
  expected_risk_reduction?: number;
  status: string;
  metadata?: RecommendationMetadata;
  generated_at: string;
  accepted_at?: string;
  completed_at?: string;
  created_at: string;
  updated_at: string;
}
